src/main/java/org/jenkinsci/plugins/gitclient/CliGitAPIImpl.java

```diff
@@ -1515,9 +1515,18 @@ private File createSshKeyFile(File key, SSHUserPrivateKey sshUser) throws IOExce
     }
 
     private String quoteWindowsCredentials(String str) {
-        // Assumes the only meaningful character is %, this may be
-        // insufficient.
-        return str.replace("%", "%%");
+        // Quote special characters for Windows Batch Files
```

Comment:
I was hesitant to change something like that just for testing purposes, but if it's thought to be worthwhile, I can certainly do so and then add some basic tests.